core/src/test/java/org/elasticsearch/index/store/ExceptionRetryIT.java

```diff
@@ -136,4 +144,58 @@ public void sendRequest(DiscoveryNode node, long requestId, String action, Trans
         assertThat(dupCounter, equalTo(0l));
         assertHitCount(searchResponse, numDocs);
     }
+
+    /**
+     * Tests that an update request does not hang in case the request fails and times out.
+     */
+    @Test
+    public void testBulkUpdateWithObserverTimeout() throws ExecutionException, InterruptedException, IOException {
```

Comment:
can we make a unit test out of this? it's TransportInstanceSingleOperationAction that's broken. Not the update request.